app/Traits/AddTagsTrait.php

```diff
@@ -2,76 +2,42 @@
 
 namespace App\Traits;
 
+use App\Actions\Photos\AddTagsToPhotoAction;
+use App\Actions\Photos\ClearTagsOfPhotoAction;
+use App\Actions\Photos\UpdateLeaderboardsFromPhotoAction;
 use App\Models\Photo;
 use App\Models\User\User;
-use App\Models\LitterTags;
-use App\Models\Location\City;
-use App\Models\Location\Country;
-use App\Models\Location\State;
-use Illuminate\Support\Facades\Redis;
 
 trait AddTagsTrait
 {
     /**
      * Add or Update tags on an image
      */
-    public function addTags ($tags, $photo_id)
+    public function addTags ($tags, $photoId)
     {
-        $photo = Photo::find($photo_id);
+        $photo = Photo::find($photoId);
         $user = User::find($photo->user_id);
 
-        $schema = LitterTags::INSTANCE()->getDecodedJSON();
+        /** @var ClearTagsOfPhotoAction $clearTagsAction */
+        $clearTagsAction = app(ClearTagsOfPhotoAction::class);
```

Comment:
I'm not sure if we want to delete all the tags before adding them again. Need to test this

Reply:
I think we need to delete them, as a user, or an admin, has no way of deleting them separately. Say, for example, a user has added smoking - cigarette butts, but it's a wrong tag. The admin now has the chance to apply new tags to the photo, and the smoking tag will be removed. If they want to keep a tag, they can add it again. I'm not sure but an admin should be able to see the current tags of a photo on the verification page. If that's true, the tag will be included on the request, so no harm is done. Let me check the admin page real quick on this.

Reply:
After a little fix, now the admin can still see the tags the user has submitted. The admin is not allowed to click the update tags button if they don't modify the photo's current tags. They can, however clear all the tags at once, I was mistaken above.

The real reason why we delete tags before applying the new ones is to easily calculate the amount of XP we should decrement from the user. We already have automated tests for this. I suggest we keep this change.